### PR TITLE
Add 5 minute buffer for expiry refresh

### PR DESCRIPTION
**Pull request by banikharbanda**

### CODE REVIEW

File: go/pkg/credshelper/credshelper.go [hunk 1]
```diff
@@ -183,7 +183,7 @@ func (ts *externalTokenSource) Token() (*oauth2.Token, error) {
 func (ts *externalTokenSource) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
 	ts.headersLock.RLock()
 	defer ts.headersLock.RUnlock()
-	if ts.expiry.Before(nowFn()) {
+	if ts.expiry.Before(nowFn().Add(-5*time.Minute)) {
```
Comment (gkousik) [inline go/pkg/credshelper/credshelper.go:186]: Can you define it as a constant at the top of the file?

Comment (banikharbanda) [reply go/pkg/credshelper/credshelper.go]: Done

File: go/pkg/credshelper/credshelper.go [hunk 2]
```diff
@@ -110,8 +112,8 @@ func buildExternalCredentials(baseCreds cachedCredentials, credsFile string, cre
 		TokenSource: oauth2.ReuseTokenSourceWithExpiry(
 			baseCreds.token,
 			baseTS,
-			// Refresh tokens 5 mins early to be safe
-			5*time.Minute,
+			// Refresh tokens a bit early to be safe
```
Comment (gkousik) [inline go/pkg/credshelper/credshelper.go:115]: super nit: this comment seems redundant :)

**Review by gkousik — APPROVED** [review]

File: go/pkg/credshelper/credshelper.go (hunk 2 above)
Comment (banikharbanda) [reply go/pkg/credshelper/credshelper.go:115]: In the next PR :)

**Pull request merged**